web/src/user_topics.js

```diff
@@ -19,45 +19,58 @@ export const visibility_policy = {
     FOLLOWED: 3,
 };
 
-export function add_muted_topic(stream_id, topic, date_muted) {
-    let sub_dict = muted_topics.get(stream_id);
-    if (!sub_dict) {
-        sub_dict = new FoldDict();
-        muted_topics.set(stream_id, sub_dict);
+export function update_user_topics(stream_id, topic, visibility_policy, date_updated) {
+    let sub_dict = all_user_topics.get(stream_id);
+    if (visibility_policy === visibility_policy.INHERIT) {
+        if (sub_dict) {
+            sub_dict.delete(topic);
+        }
+    } else {
+        if (!sub_dict) {
+            sub_dict = new FoldDict();
+            all_user_topics.set(stream_id, sub_dict);
+        }
+        const time = get_time_from_date_muted(date_updated);
+        sub_dict.set(topic, {date_updated: time, visibility_policy});
     }
-    const time = get_time_from_date_muted(date_muted);
-    sub_dict.set(topic, time);
 }
 
-export function remove_muted_topic(stream_id, topic) {
-    const sub_dict = muted_topics.get(stream_id);
-    if (sub_dict) {
-        sub_dict.delete(topic);
+export function get_topic_visibility_policy(stream_id, topic) {
+    if (stream_id === undefined) {
+        return false;
     }
+    const sub_dict = all_user_topics.get(stream_id);
+    if (sub_dict && sub_dict.get(topic)) {
+        return sub_dict.get(topic).visibility_policy;
+    }
+
+    return visibility_policy.INHERIT;
+}
+
+export function is_topic_unmuted(stream_id, topic) {
```

Comment:
Instead of using both `is_topic_unmuted` and `is_topic_muted`, we can use a single function named `topic_has_visibility_policy`.

You can take a look at the [corresponding backend function](https://github.com/zulip/zulip/blob/main/zerver/lib/user_topics.py#L180) for your reference.

This approach saves the work of adding a new function every time a new visibility_policy is added. e.g. `is_topic_followed`.

Reply:
Sure, I will add a common function in the updated PR.